guava/src/com/google/common/hash/AbstractByteHasher.java

```diff
@@ -63,7 +71,9 @@ protected void update(ByteBuffer b) {
   }
 
   /** Updates the sink with the given number of bytes from the buffer. */
-  private Hasher update(int bytes) {
+  @SuppressWarnings({"upperbound:argument.type.incompatible","lowerbound:argument.type.incompatible"})// Since `scratch.array().length` will return 8, `bytes`
+  // range from 0 to 8) - 1 < scratch.array().length
```

Comment:
Can you add a pointer in this comment to why we know that scratch is always length 8? Further, if that's so then it should be declared as `@ArrayLen(8)` - the typechecker ought to be able to infer that the code is correct once that's done. If it doesn't, you can file an issue.